stand/efi/loader/efi_main.c

```diff
@@ -87,6 +87,15 @@ efi_main(EFI_HANDLE image_handle, EFI_SYSTEM_TABLE *system_table)
 	BS = ST->BootServices;
 	RS = ST->RuntimeServices;
 
+#ifdef __i386__
+	/* Check for long mode */
+	asm("cpuid" : "=d"(status) : "a"(0x80000001) : "%ebx", "%ecx");
+	if (!(status & 0x20000000)) {
+		ST->ConOut->OutputString(ST->ConOut, (CHAR16 *)L"No long mode.\r\n");
```

Comment:
I'd wrap the message, and expaned the message to be 'No long mode support on this CPU.' Maybe with a 'Unable to proceed'

Reply:
Not sure what you mean by wrap, but I've changed it to: `This CPU doesn't support long mode.\r\nUnable to proceed.\r\n`